mobilecoinfoundation/sgx
# Fix docs builds w/o SGX SDK installed

- Stub out panic crate.
- Add dependabot, rustfmt, audit, and the impl.
- Enable LTO, remove stale rustfmt ignores.
- Move trusted code into dedicated directory
- Verify the quoting enclave report body
- Fix misspelling of ASN.1 in function name
- Add README.md for quote verification
- Rename `get_asn1_signature()` to `get_der_signature()`
- Consistently formatted line lengths to 80
- Add FFI for `sgx_status_t`
- Remove empty `mc-sgx-core-sys` crate
- Use `core:ffi` for the ctypes prefix
- Add ability to verify enclave report body
- Remove extraneous comments
- Verify the attestation key
- Add `Display` to the verification Error
- Add check to ensure zero bytes after attestation key digest
- Add rust interface to untrusted SGX
- Add github actions to run untrusted tests
- Use indentation in yaml instead of braces
- Fix cargo fmt that came in from last merge
- Add FFI interface for trts(trusted runtime system)
- Add `mc-sgx-core-build` crate
- Add newlines between external and local crate dependency groups
- Remove sections from TOC that don't exist
- Add FFI bindings for SGX tcrypto library
- Add missing newlines
- Specify `mc-sgx-crypto-sys*` as `no_std`
- Add FFI bindings to SGX tservice library
- Remove test check from `no_std` usage
- github actions now check for lock file changes
- Add `#![no_std]` to the mc-sgx-service-sys* crate
- Simplify `sgx` prefix for core types
- Utilize RustCrypto for SecP256R1
- Update to nightly-2022-07-22
- Verify the quote certificate chain
- Add note on documentation location for certificate order
- Update the lock file for new dependencies
- Consolidate workspaces into one
- Rename `sys` crates to `ffi`
- Add Cargo nono to the ci lint check
- WIP
- Add capable-ffi-types
- Finish off capable/ffi/types
- Add start of mc-sgx-capable-ffi-sys crate.
- Build Cleanup
- Update README for ffi-sys crate.
- Change -sys to use static linkage.
- Use static linkage for sgx-capable FFI.
- Fix license to Apache-2.0
- Consolidate ParseCallbacks and Builder to common
- Add markdownlint as a lint step on the CI
- Fix dependabot for directory layout
- Use &str slice for core types
- Rename crate from FFI to SYS
- Fix dependabot for directory layout
- Add markdownlint as a lint step on the CI
- Consolidate ParseCallbacks and Builder to common
- Use &str slice for core types
- Make urts functions explicit
- Rename capapble-ffi to capable-sys.
- Format and fix ffi refs.
- Apply suggestions from code review
- Relax markdown rules, nono ignore capable-sys.
- Rename `crypto`->`tcrypto` and `service`->`tservice`
- Fiddle with the markdownlint run
- Add CHANGELOG.md for tracking changes
- Make tservice functions explicit
- Make tcrypto functions explicit
- Add missing `sgx_kss_config_t`
- Remove private functions
- Adding missing header for tservice crate
- Add some missing types from sgx_report2.h
- Disable generating comments in bindings
- Add tstdc FFI
- Add `static` qualifier to c library link flag
- Convert to `sim` feature for SGX crates
- Make trts functions explicit
- Remove binding for private function
- WIP
- Add dcap_ql FFI
- Sort package section, add keywords and categories
- Fix links in tstdc
- Add missing types from sgx_pce header
- Fix cargo clippy warning
- Add dcap_quoteverify FFI
- Apply suggestions from code review
- Add missing dcap quoteverify Cargo.toml keys
- Exclude `mc-sgx-dcap-quoteverify-sys` from cargo nono check
- Move types from `sgx_quote.h` to `mc-sgx-core-sys-types`
- Add missing DCAP ql Cargo.toml keys
- Exclude mc-sgx-dcap-ql-sys from cargo nono check
- Fix link to ParseCallbacks
- Add sgx_include_path() method, make sgx_library_path() return just the lib.
- Update core-sys-types to reflect core-build changes.
- Update dcap-ql to handle changes in core-build.
- Update dcap_quoteverify to match core-build, remove spurious linker flags from dcap_ql
- adding the dcap headers
- Fix location of manifest dir, updates to support core-build
- Update tservice to handle core-build changes.
- Update various crate links
- Update tstdc for core-build changes
- Fix bare URLs in quote_verify
- Add additional intel headers
- Remove superfluous allowlist_recursively().
- Update urts-sys-foo to handle core-build changes.
- Fix tstdc attr stuff.
- Simplify path functions.
- Port capable to use core-build
- Fix paths in tcrypto
- Add doc job to build workflow.
